pkg/reconciler/helper.go

```diff
@@ -0,0 +1,61 @@
+package reconciler
+
+import (
+	"k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	appsv1listers "k8s.io/client-go/listers/apps/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+// ReconcileDeployment reconciles the K8s Deployment 'd'.
+func ReconcileDeployment(kubeClient kubernetes.Interface, lister appsv1listers.DeploymentLister, d *v1.Deployment) (*v1.Deployment, error) {
```

Comment:
This should take an event recorder or return an event to record. An event should be recorded whenever a deployment is created or updated.